src/main/java/org/opentripplanner/service/vehiclerental/internal/DefaultVehicleRentalService.java

```diff
@@ -114,7 +114,9 @@ public List<VehicleRentalPlace> getVehicleRentalStationForEnvelope(
     return rentalPlaces
       .values()
       .stream()
+      .filter(VehicleRentalStation.class::isInstance)
       .filter(b -> envelope.contains(new Coordinate(b.getLongitude(), b.getLatitude())))
+      .map(VehicleRentalStation.class::cast)
       .toList();
```

Comment:
I wonder if it's worth it to not reuse the getVehicleRentalStations methods here or alternatively reusing some private method that returns a stream of the vehicle rental stations for performance reasons instead of doing this filter/map code here again?

Reply:
done